resources/js/Pages/Dashboard.jsx

```diff
@@ -0,0 +1,222 @@
+import AuthenticatedLayout from "@/Layouts/AuthenticatedLayout";
+import { Head } from "@inertiajs/react";
+import axios, * as others from "axios";
+import { useState, useRef } from "react";
+import Card from "react-bootstrap/Card";
+import { useMemo, useEffect } from "react";
+import { debounce } from "lodash";
+import { Navbar } from "react-bootstrap";
+import "./../../css/custom.css";
+// import { Container } from "postcss";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+
+const Dashboard = ({ auth, accessToken }) => {
+    const [isLoad, setIsLoad] = useState(false);
+    const [isUniSelected, setIsUniSelected] = useState(false);
+    const [noDataFound, setNoDataFound] = useState(false);
+
+    const timer = null;
+
+    const [arrData, setArrData] = useState([]);
+    const [selectedText, setSelectedText] = useState({});
+
+    const [searchQuery, setSearchQuery] = useState("");
+    const ref = useRef(null);
+
+    const searchInApi = async (event) => {
+        try {
+            setArrData([]);
+            setIsUniSelected(false);
+            setNoDataFound(false);
+            if (event.target.value.length >= 3) {
```

Comment:
Here I am already managing data into `setArrData `state, so I do not need to `setNoDataFound `state to handle not found. I can handle it not found data using `ArrData `length and for `setIsUniselected `was used because to show searched result in details but I do not need to it I can manage it using `selectedText`.